### PR TITLE
Add secure setting for watcher email password

### PR DESCRIPTION
**Pull request by hub-cap**
Other watcher actions already account for secure settings in their
sensitive settings, whereas the email sending action did not. This adds
the ability to optionally set a secure_password for email accounts.

### CODE REVIEW

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java [hunk 1]
```diff
@@ -232,7 +254,10 @@ static Properties loadSmtpProperties(Settings settings) {
                 settings = builder.build();
                 Properties props = new Properties();
                 for (String key : settings.keySet()) {
```
Comment (spinscale) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: how about using `settings.filter(s -> s.startsWith("_secure") == false).keySet()`?

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java [hunk 2]
```diff
@@ -204,13 +209,30 @@ public Session createSession() {
...
+                String passStr = getNullableSetting(SMTP_PASSWORD, settings, SECURE_PASSWORD_SETTING);
                 password = passStr != null ? passStr.toCharArray() : null;
                 properties = loadSmtpProperties(settings);
             }
 
+            /**
+             * Finds a setting, and then a secure setting if the setting is null, or returns null if one does not exist. This differs
+             * from other getSetting calls in that it allows for null whereas the other methods throw an exception.
+             */
+            private static String getNullableSetting(String settingName, Settings settings, Setting<SecureString> secureSetting) {
```
Comment (spinscale) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: the name is not too descriptive but I dont have a good alternative at the moment either

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java [hunk 3]
```diff
@@ -204,13 +209,30 @@ public Session createSession() {
 
             Smtp(Settings settings) {
                 host = settings.get("host", settings.get("localaddress", settings.get("local_address")));
+
                 port = settings.getAsInt("port", settings.getAsInt("localport", settings.getAsInt("local_port", 25)));
                 user = settings.get("user", settings.get("from", null));
-                String passStr = settings.get("password", null);
+                String passStr = getNullableSetting(SMTP_PASSWORD, settings, SECURE_PASSWORD_SETTING);
```
Comment (spinscale) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: how about storing the password as a `SecureString` and have `getNullableSettings` return `SecureString` as well?

**Review by spinscale — CHANGES_REQUESTED** [review]
can you add a test as well?

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java (hunk 3 above)
Comment (hub-cap) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: done.

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java (hunk 1 above)
Comment (hub-cap) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: +1

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java (hunk 2 above)
Comment (hub-cap) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: yea was only cuz the other classes have a method like this called `getSetting` that throws up on null :)

**Comment by hub-cap** [comment]
test added @spinscale

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java [hunk 4]
```diff
@@ -24,10 +27,13 @@
 import java.security.PrivilegedActionException;
 import java.security.PrivilegedExceptionAction;
 import java.util.Properties;
+import java.util.Set;
 
 public class Account {
 
     static final String SMTP_PROTOCOL = "smtp";
+    static final String SMTP_PASSWORD = "password";
```
Comment (spinscale) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Account.java]: can be made private?

**Review by spinscale — APPROVED** [review]
LGTM. can you add the commit line to sth like `Watcher: Allow email account passwords to be a secure setting`

**Pull request merged**